src/interfaces/collection.ts

```diff
@@ -8,6 +8,7 @@ export interface ICollection<T> {
   findByProperties(properties: Partial<T>): T;
   forEach(callback: (object: T, index?: number) => void): void;
   mapByProperty<P extends keyof T>(property: P): (T[P])[];
+  keyByProperty<P extends keyof T>(property: P): { [p: string]: T; };
```

Comment:
Typescript allows bracket property-access, no matter if we restrict to `number` or `string` in `[p: string]`. See sample below:

Reply:
I've alway thought that this is a perfect use case for `Map`, which would preserve the value type. WDYT @nicolasgramlich ?

Reply:
Hm, I guess I'm too much of a Lodash fanboy and used to the syntax. 😆 

I actually started with a Map, but decided to not convert from the Lodash object result to a Map.

I'd say 99% of the time the use case is the Lodash way, isn't it?

Maybe we could have both `keyBy` (Map Version) and `keyByProperty` (Lodash version).

Reply:
Alright, alright, let's lodash this!